regression/pages/studio/studio_home.py

```diff
@@ -31,14 +31,10 @@ def select_course(self, course_title):
         """
         Selects the course we want to perform tests on
         """
-        course_title_selector = '.course-link h3'
-        self.wait_for_element_presence(course_title_selector,
-                                       'Course link presence')
-        course_names = self.q(css=course_title_selector)
-        for vals in course_names:
-            if course_title in vals.text:
-                vals.click()
-                return
+        elements = self.browser.find_elements_by_xpath('//h3[contains(text(), "%s")]' % course_title)
```

Comment:
You could use `self.q(xpath='//h3[contains(text(), "{}")]'.format(course_title)).first.click()` to get promises and retries built in so that you don't get stale element exceptions.